doc/nrf/releases/release-notes-changelog.rst

```diff
@@ -516,6 +516,7 @@ Libraries for networking
 
     * Updated timeout handling. Now using http_client library timeout also.
     * Removed CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT and CONFIG_REST_CLIENT_SCKT_RECV_TIMEOUT.
+    * Updated to handle a zero timeout value as "no timeout" to avoid immediate timeouts.
```

Comment:
This is not clear to me what sense of "no timeout" you mean.  Wait forever?  Or return with something like -EAGAIN rather than -ETIMEDOUT?